Remove exp10 in TH (unused)

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30422 Remove exp10 in TH (unused)**
* #30329 Remove many unused bfloat16 functions in TH
* #30176 Migrate reciprocal from the TH to Aten (CUDA)

Differential Revision: [D18764280](https://our.internmc.facebook.com/intern/diff/D18764280)

xuhdev committed:
Remove exp10 in TH (unused)

[ghstack-poisoned]

xuhdev committed:
Update on "Remove exp10 in TH (unused)"




[ghstack-poisoned]

xuhdev committed:
Update on "Remove exp10 in TH (unused)"




[ghstack-poisoned]

xuhdev committed:
Update on "Remove exp10 in TH (unused)"


Differential Revision: [D18764280](https://our.internmc.facebook.com/intern/diff/D18764280)

[ghstack-poisoned]

xuhdev commented:
@pytorchbot merge this please

VitalyFedyunin commented:
Please rebase to master, last night I wasn't able to land due to failing on master, but now it is not in sync.

xuhdev committed:
Update on "Remove exp10 in TH (unused)"


Differential Revision: [D18764280](https://our.internmc.facebook.com/intern/diff/D18764280)

[ghstack-poisoned]

xuhdev commented:
@VitalyFedyunin Rebased
